File: Backend/AstroGame.Shared/Models/Technologies/FinishedTechnologies/StellarObjectDependentFinishedTechnology.cs
using System;
using AstroGame.Shared.Models.Players;

namespace AstroGame.Shared.Models.Technologies.FinishedTechnologies
{
    public abstract class StellarObjectDependentFinishedTechnology : FinishedTechnology
    {
        public Guid ColonizedStellarObjectId { get; set; }

        public virtual ColonizedStellarObject ColonizedStellarObject { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/finished_technologies/stellar_object_dependent_finished_technology.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/common/guid.dart';
import 'package:astrogame_app/models/finished_technologies/finished_technology.dart';

import 'package:json_annotation/json_annotation.dart';

@GuidConverter()
abstract class StellarObjectDependentFinishedTechnology
    extends FinishedTechnology {
  @JsonKey()
  late Guid colonizedStellarObjectId;

  StellarObjectDependentFinishedTechnology();
}
